middlewares/access.middleware.js

```diff
@@ -0,0 +1,190 @@
+                case 'VENDOR_ADDITIONAL':
+                    if (res.locals.userData.access.vendor_next_payment_date < now) {
+                        errorResponse.send(res, 403, 'Vendor Access Required');
+                    } else if (res.locals.userData.access.vendor_additional_paid === true) {
+                        errorResponse.send(res, 409, 'Vendor Additional Already Paid');
+                    }else {
+                        res.locals.additional = true;
+                        next();
+                    }
+                    break;
+                default:
+                    errorResponse.send(res, 400, 'Malformed Request');
+                    break;
+            }
+        } catch (err) {
+            errorResponse.send(res, 500, 'Internal Server Error', err);
+        }
+    }
+
+    function createSubscription(req, res, next) {
+        if (res.locals.vendor === true || res.locals.customer === true) {
+            stripe.subscriptions.create({
+              customer: res.locals.userData.stripe_id,
+              items: [
+                {
+                  plan: req.body.type.type,
+                },
+              ],
+              source: req.body.token.id
+            }, function(err, subscription) {
+                if (err) {
+                    errorResponse.send(res, 500, 'Internal Server Error', err);
+                } else {
+                    if (res.locals.vendor === true) {
+                        res.locals.userData.access.vendor_payment_amount = subscription.plan.amount;
+                        res.locals.userData.access.vendor_next_payment_date = new Date(subscription.current_period_end * 1000);
+                        next();
+                    } else if (res.locals.customer === true) {
+                        res.locals.userData.access.customer_payment_amount = subscription.plan.amount;
+                        res.locals.userData.access.customer_next_payment_date = new Date(subscription.current_period_end * 1000);
+                        next();
+                    } else {
+                        errorResponse.send(res, 500, 'Internal Server Error');
+                    }
+                }
+            });
+        } else {
+            next();
+        }
+    }
+
+    function createCharge(req, res, next) {
+        if (res.locals.additional === true) {
+            stripe.charges.create({
+                amount: 300,
+                currency: "usd",
+                description: "Vendor Additional Charge",
+                source: req.body.token.id,
+            }, function(err, charge) {
+                if (err) {
+                    errorResponse.send(res, 500, 'Internal Server Error', err);
+                } else {
+                    res.locals.userData.access.vendor_additional_paid = true;
+                    next();
+                }
+            });
+        } else {
+            next();
+        }
+    }
+
+    function updateUserEntity(req, res) {
```

Comment:
I think we also need to update the user's last created property status
b/c user creates property and then make the payment. Property should be shown only after making payment

Reply:
isn't user pay first then creates property? otherwise we wont be checking access before creating property